gef.py

```diff
@@ -4661,8 +4661,8 @@ def add_setting(self, name: str, value: Tuple[Any, type, str], description: str
 
     def __setitem__(self, name: str, value: Union[Any, Tuple[Any, str]]) -> None:
         # make sure settings are always associated to the root command (which derives from GenericCommand)
-        if "GenericCommand" not in [x.__name__ for x in self.__class__.__bases__]:
-            return
+        #if "GenericCommand" not in [x.__name__ for x in self.__class__.__bases__]:
```

Comment:
why this change?

Reply:
I'd actually forgotten to look at that closer before sending the PR...  

self.__class__.__bases__ only contains the immediate parent classes, and that meant that GotAuditCommand wouldn't have any settings because GenericCommand isn't in that list.  I think I can revert this, and just add GenericCommand as a second parent class in GotAuditCommand (which will look *super* weird to future maintainers)

Reply:
That is, GotAuditCommand doesn't provide an `__init__`, so GotCommand's runs.  GotCommand's `__init__` should set values for function_resolved and function_not_resolved, but those don't actually get set because GenericCommand isn't an immediate parent of the class that's actually being instantiated, and the instance ends up with no settings at all.

I'm not sure I understand the intent of this check, and maybe checking isinstance() instead would be better?